src/meta/src/hummock/compaction/picker/tier_compaction_picker.rs

```diff
@@ -50,12 +50,134 @@ impl TierCompactionPicker {
         // do not pick the first sub-level because we do not want to block the level compaction.
         let non_overlapping_type = LevelType::Nonoverlapping as i32;
         for (idx, level) in l0.sub_levels.iter().enumerate() {
-            if level.level_type == non_overlapping_type
-                && level.total_file_size > self.config.sub_level_max_compaction_bytes
+            if level.level_type != non_overlapping_type
+                || level.total_file_size > self.config.sub_level_max_compaction_bytes
             {
                 continue;
             }
 
+            let mut table_id = 0;
+            let mut several_table = false;
+            for sst in &level.table_infos {
+                if sst.table_ids.len() > 1 {
+                    several_table = true;
+                    break;
+                }
+                if level_handler.is_pending_compact(&sst.id) {
+                    continue;
+                }
+                table_id = sst.table_ids[0];
```

Comment:
Why do we choose the table_id of the last sst as the identifier for this task? The same concern whether starvation will occur